ipatests/test_webui/test_selfservice.py

```diff
@@ -49,4 +59,230 @@ def test_crud(self):
+
+        # try to add peemission without name but having attribute
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.wait_for_request()
+        self.facet_button_click('add')
+        self.check_option('attrs', 'displayname')
+        self.dialog_button_click('add')
+        self.assert_field_validation(FIELD_REQ, field='aciname')
+        self.dialog_button_click('cancel')
+
+        # try to add peemission without having attribute
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.wait_for_request()
+        self.facet_button_click('add')
+        self.fill_textbox('aciname', data_selfservice.DATA1['pkey'])
+        self.dialog_button_click('add')
+        self.assert_field_validation(FIELD_REQ, field='attrs')
+        self.dialog_button_click('cancel')
+
+        # try to add aciname with leading space
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.wait_for_request()
+        self.facet_button_click('add')
+        self.fill_textbox('aciname', ' %s' % data_selfservice.DATA1['pkey'])
+        self.check_option('attrs', 'audio')
+        self.dialog_button_click('add')
+        self.assert_last_error_dialog(INV_NAME)
+        self.dialog_button_click('cancel')
+        self.dialog_button_click('cancel')
+
+        # try to add aciname with trailing space
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.wait_for_request()
+        self.facet_button_click('add')
+        self.fill_textbox('aciname', '%s ' % data_selfservice.DATA1['pkey'])
+        self.check_option('attrs', 'audio')
+        self.dialog_button_click('add')
+        self.assert_last_error_dialog(INV_NAME)
+        self.dialog_button_click('cancel')
+        self.dialog_button_click('cancel')
+
+        # try to add aciname with special char
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.wait_for_request()
+        self.facet_button_click('add')
+        self.fill_textbox('aciname', '#%^')
+        self.dialog_button_click('add')
+        self.assert_field_validation(ERR_INCLUDE, field='aciname')
+        self.dialog_button_click('cancel')
+
+        # try to modify pesmission by removing all attributes
+        self.add_record(data_selfservice.ENTITY, data_selfservice.DATA1)
+        self.navigate_to_record(data_selfservice.DATA1['pkey'])
+        self.check_option('attrs', 'businesscategory')
+        self.facet_button_click('save')
+        self.assert_field_validation(FIELD_REQ, field='attrs')
+
+    @screenshot
+    def test_del_multiple_permission(self):
+        """
+        Try to delete multiple self service permission
+        """
+        self.init_app()
+        self.add_record(data_selfservice.ENTITY, data_selfservice.DATA1)
+        self.add_record(data_selfservice.ENTITY, data_selfservice.DATA)
+        self.delete_record([data_selfservice.DATA1['pkey'],
+                            data_selfservice.DATA['pkey']])
```

Comment:
Would be good to add some asserts, e.g `assert_record(xy, negative=True) after `delete_record` call that they are deleted because `delete_record will happily not do anything if e.g. you are on a different page.